src/stlink-lib/usb.c

```diff
@@ -1172,7 +1172,7 @@ stlink_t *stlink_open_usb(enum ugly_loglevel verbose, enum connect_type connect,
 
         if (ret) { continue; } // could not open device
 
-        uint32_t serial_len = stlink_serial(handle, &desc, sl->serial);
+        uint32_t serial_len = (uint32_t)stlink_serial(handle, &desc, sl->serial);
```

Comment:
I would simply change the type of `serial_len` variable to `size_t`.

Reply:
We could alternatively also change stlink_serial() to uint32_t.
This should be safe on all common architectures as well.